File: scala/src/bron_kerbosch2_g.scala
import bron_kerbosch_pivot.PivotChoice.MaxDegree
import bron_kerbosch_pivot.visit

object bron_kerbosch2_g extends bron_kerbosch_algorithm {
  def explore(graph: UndirectedGraph): Cliques = {
    val candidates = graph.connected_vertices().toSet
    if (candidates.nonEmpty) {
      visit(graph, MaxDegree, MaxDegree, candidates, Set(), Seq())
    } else {
      Seq()
    }
  }
}

File: python3/bron_kerbosch2_g.py
# coding: utf-8

from bron_kerbosch_pivot import pick_max_degree, visit
from graph import UndirectedGraph
from reporter import Reporter


def bron_kerbosch2_g(graph: UndirectedGraph, reporter: Reporter) -> None:
    """Bron-Kerbosch algorithm with pivot of highest degree"""
    if candidates := graph.connected_vertices():
        visit(graph=graph,
              reporter=reporter,
              initial_pivot_choice=pick_max_degree,
              further_pivot_choice=pick_max_degree,
              candidates=candidates,
              excluded=set(),
              clique=[])
